### PR TITLE
feat: verify OAuth scopes of classic GitHub PATs

### PR DESCRIPTION
**Pull request by jedwards1211**
Solves #895 for classic PATs only

### CODE REVIEW

File: test/integration.test.js [hunk 1]
```diff
@@ -949,7 +967,10 @@ test("Verify and notify failure", async (t) => {
     .get(
       `https://api.github.local/repos/${owner}/${repo}`,
       {
-        permissions: { push: true },
+        permissions: {
+          push: true,
+          maintain: true,
```
Comment (babblebey) [inline test/integration.test.js:972]: @jedwards1211

Is `maintain` always `true`?? 🤔

Comment (jedwards1211) [reply test/integration.test.js:972]: No but users for whom it's false may not be able to write issues, releases and pull requests, so I added a check that it's true in this PR.

Comment (jedwards1211) [reply test/integration.test.js:972]: The props in permissions seem to be based upon this: https://github.nih.gov/about/features/access-permissions#:~:text=Maintain%3A%20Recommended%20for%20project%20managers,pull%20requests%20without%20write%20access

Comment (babblebey) [reply test/integration.test.js:972]: Sounds great, I'll give this a whirl 🛞😁

So, how about a coverage for case when `maintain` is `false`, let's do that!? 😉

Comment (jedwards1211) [reply test/integration.test.js:972]: There's a case for that in [verify.test.js](https://github.com/semantic-release/github/pull/897/files#diff-8baee4e69982184abf900a755dc4ca1bffe8acfc174b826186bc184d07b3187cR1447) now, do you want one in integration.test.js?

Comment (babblebey) [reply test/integration.test.js:972]: Oh yes please 😁

**Review by babblebey — APPROVED** [review]
This looks great @jedwards1211 

Thank you 🫶 

Though waiting on the test coverage, but I'm approving this yea 😁

File: test/integration.test.js (hunk 1 above)
Comment (jedwards1211) [reply test/integration.test.js:1009]: Okay is this sufficient? https://github.com/semantic-release/github/pull/897/commits/f5521226c5736d7b8c56a59d8249fd85a712b6cb#diff-6bb01a9664f0d2819bc4e2290b685dddc6dfb10fd786e7fa30933ae5ec57212bR74

Comment (babblebey) [reply test/integration.test.js:1009]: Let's gooooooo 💪

**Comment by babblebey** [comment]
Sure thing @jedwards1211.... and yooooo we noticed you've been very active around our project for some time now and your contributions are of great value.

We'd therefore like to invite you as an additional maintainer for the GitHub plugin (this way you can surely get the pings auto-magically haha 😃)

Would that be something you're interested in??? 😉 

Cc: @gr2m

**Comment by jedwards1211** [comment]
Yeah that sounds good!  Do y'all have any kind of meetings to plan things together?

**Pull request merged**